assets/js/atomic/blocks/product/sku/block.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import {
+	useInnerBlockLayoutContext,
+	useProductDataContext,
+} from '@woocommerce/shared-context';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+/**
+ * Product SKU Block Component.
+ *
+ * @param {Object} props             Incoming props.
+ * @param {string} [props.className] CSS Class name for the component.
+ * @param {Object} [props.product]   Optional product object. Product from context will be used if
+ *                                   this is not provided.
+ * @return {*} The component.
+ */
+const Block = ( { className, ...props } ) => {
+	const { parentClassName } = useInnerBlockLayoutContext();
+	const productDataContext = useProductDataContext();
+	const product = props.product || productDataContext.product || {};
+	const sku = product.sku || '';
+
+	if ( ! sku ) {
+		return null;
+	}
+
+	return (
+		<div
+			className={ classnames(
+				className,
+				'wc-block-components-product-sku',
+				`${ parentClassName }__product-sku`
```

Comment:
I'm coming in late on the class names discussion .. keen to understand why we need both classes here. Not a blocker, just something I need to get up to speed on.

If having both classes with parent prefix here is a common pattern, perhaps we can wrap `classnames` and make our own little utility for atomic blocks. Then we can document the expectations/rationale and ensure all atomic blocks do it in a consistent way. (Not a blocker just thinking out loud.)

Reply:
We did want to drop the prefixed ones but it's kept to avoid breaking compatibility with all products block. Now it's just here for consistency with the other atomic blocks.

Some of the suffixes differ though. We can maybe unify once we get all atomic blocks added.